osu.Game/Rulesets/Mods/ModTimeRamp.cs

```diff
@@ -59,23 +58,8 @@ private void applyAdjustment(double amount)
         {
             double adjust = 1 + (Math.Sign(FinalRateAdjustment) * Math.Clamp(amount, 0, 1) * Math.Abs(FinalRateAdjustment));
 
-            switch (clock)
-            {
-                case IHasPitchAdjust pitch:
-                    pitch.PitchAdjust /= lastAdjust;
-                    pitch.PitchAdjust *= adjust;
-                    break;
-
-                case IHasTempoAdjust tempo:
-                    tempo.TempoAdjust /= lastAdjust;
-                    tempo.TempoAdjust *= adjust;
-                    break;
-
-                default:
-                    clock.Rate /= lastAdjust;
-                    clock.Rate *= adjust;
-                    break;
-            }
+            track.Tempo.Value /= lastAdjust;
+            track.Tempo.Value *= lastAdjust;
```

Comment:
Am I missing something, or should this be
```suggestion
            track.Tempo.Value *= adjust;
```
I haven't run this to be fair, but looks suspicious on my eye test of the diff.

Reply:
good catch. this ugly code won't exist in the follow-up PR, thankfully.